### PR TITLE
Add EMR streaming job launcher

### PR DESCRIPTION
**Pull request by oavdeev**
**What this PR does / why we need it**:
Add streaming ingestion job support to the EMR launcher and the launcher interface

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/pyspark/launcher.py [hunk 1]
```diff
@@ -99,16 +102,19 @@ def _source_to_argument(source: DataSource):
         "date_partition_column": source.date_partition_column,
     }
 
-    kind, option_field = _SOURCES[type(source)]
```
Comment (oavdeev) [inline sdk/python/feast/pyspark/launcher.py:102]: `option_field` here was unused anyway after #1049

File: sdk/python/feast/pyspark/launchers/aws/emr.py [hunk 2]
```diff
@@ -227,7 +239,32 @@ def offline_to_online_ingestion(
 
         job_ref = self._submit_emr_job(step)
 
-        return EmrIngestionJob(self._emr_client(), job_ref)
+        return EmrBatchIngestionJob(self._emr_client(), job_ref)
+
+    def start_stream_to_online_ingestion(
```
Comment (woop) [inline sdk/python/feast/pyspark/launchers/aws/emr.py:244]: Can we please get a docstring here?

**Review by pyalex — APPROVED** [review]

**Pull request merged**